bare_metal_billing/models.py

```diff
@@ -0,0 +1,55 @@
+from decimal import Decimal
+from typing import Annotated
+import datetime
+
+import pydantic
+
+
+BM_UNKNOWN_SU = "BM UNKNOWN"
+
+
+def parse_date(v: str | datetime.date) -> datetime.date:
+    if isinstance(v, str):
+        return datetime.datetime.fromisoformat(v)
+    return v
+
+
+PositiveDecimalField = Annotated[Decimal, pydantic.Field(gt=0)]
+PositiveIntField = Annotated[int, pydantic.Field(gt=-1)]
+DateField = Annotated[datetime.datetime, pydantic.BeforeValidator(parse_date)]
+
+
+class BMNodeUsage(pydantic.BaseModel):
+    resource_class: Annotated[str, pydantic.Field(alias="Resource Class")]
+    project: Annotated[str, pydantic.Field(alias="Project")]
+    start_time: Annotated[DateField, pydantic.Field(alias="Start Time")]
+    expire_time: Annotated[DateField | None, pydantic.Field(alias="Expire Time")] = None
+
+
+class BMUsageData(pydantic.RootModel):
+    root: list[BMNodeUsage]
+
+    @pydantic.model_validator(mode="after")
+    def validate_expire_time(self):
+        validated_list = []
+        for node_usage in self.root:
+            if node_usage.expire_time:
+                if node_usage.expire_time < node_usage.start_time:
+                    continue
```

Comment:
So, we just ignore any lease info where the expire_time is before the start_time? Shouldn't that be unexpected and worth raising an error?

Reply:
also, what if expire_time == start_time?